toptal/picasso
# feat(RatingThumbs): add new component for rating positive negative with thumbs

[FX-2468](https://toptal-core.atlassian.net/browse/FX-2468)

### Description

Following the descriptions in [figma](https://www.figma.com/file/5SCTOPrCDcHuk5We091GBn/Product-Library?node-id=66%3A0) and examples in abstract [1](https://share.goabstract.com/a94bac67-e89d-457f-92e9-252aa1de21c8?mode=design&sha=b936c32f6702f89e571cd59b0cf84af8bf8de9fa) [2](https://share.goabstract.com/3f1af96b-73af-4471-b049-6340ddc480db?mode=design&sha=b936c32f6702f89e571cd59b0cf84af8bf8de9fa), we are adding a new component for positive and negative ratings. Following the [decision on either create a new component or extend the current one](https://toptal-core.atlassian.net/wiki/spaces/FE/pages/2614165609/Should+we+extend+Rating+component+with+thumbs+variant+or+create+ThumbsRating), I created a new component called RatingThumbs that has a boolean state (`true` for a positive rating and `false` for a negative rating), an `undefined` state means unselected.

The storybook examples were added to the Rating page, as requested

### How to test

- Check new [component and examples on storybook](https://picasso.toptal.net/FX-2468-add-new-component-rating-thumbs/?path=/story/components-rating--rating) (at the end of the page for the Rating component)

### Screenshots

New component:
![image](https://user-images.githubusercontent.com/5845160/153922267-eecc54a6-0202-4613-9182-587c7961c849.png)

Default state with hover effect on the right:
![image](https://user-images.githubusercontent.com/5845160/153922349-030a08c7-73a1-44ef-84eb-61e319d37827.png)


### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
